public/js/menu.js

```diff
@@ -50,4 +50,17 @@ window.onload = () => {
     }
 
     updateFilterNumbers();
+
+    window.onscroll = function () {
+        const categoryLogos = document.querySelectorAll('.flipable');
+        categoryLogos.forEach((element) => {
+            const observer = new IntersectionObserver(
+                (entries) => {
+                    element.classList.toggle('flip', entries[0].isIntersecting);
+                },
+                { threshold: 1 }
+            );
+            observer.observe(element);
+        });
+    };
```

Comment:
I believe this might be a bit of a memory leak 😬 For every px scrolled, a new intersection observer is created, which is then used to observe multiple elements. I believe we can simply drop the onscroll function wrapper here completely 🙂

Reply:
Hmm bit more complicated than that even! Apparently the rotate3d of the animation causes the intersection to re-calculate, so the moment the animation is at the exact 90 degree spot, the `isIntersection` is back to `false`, causing a loop! 😵

Reply:
Added a solution in https://github.com/cmdmethods/cmdmethods/pull/42 🙂